tests/testAirPressure.cpp

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: GPL-2.0
+#include "testAirPressure.h"
+#include "core/divesite.h"
+#include "core/divelist.h"
+#include "core/file.h"
+#include <QString>
+#include <core/qthelper.h>
+
+void TestAirPressure::initTestCase()
+{
+	/* we need to manually tell that the resource exists, because we are using it as library. */
+	Q_INIT_RESOURCE(subsurface);
+}
+
+void TestAirPressure::get_dives()
+{
+	struct dive *dive;
+	verbose = 1;
+
+	QCOMPARE(parse_file(SUBSURFACE_TEST_DATA "/dives/TestAtmPress.xml", &dive_table, &trip_table, &dive_site_table), 0);
+	dive = get_dive(0);
+	dive->selected = true;
+	QVERIFY(dive != NULL);
+}
+
+void TestAirPressure::testReadAirPressure()
+{
+	struct dive *dive;
+	dive = get_dive(0);
+	dive->selected = true;
+	QVERIFY(dive != NULL);
```

Comment:
Well, maybe we should do this test before dereferencing `dive` in the line above?